scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -164,10 +164,24 @@ class Router(formatOps: FormatOps) {
       // Interpolated string left brace
       case FormatToken(open @ T.LeftBrace(), _, _)
           if leftOwner.is[SomeInterpolate] =>
-        Seq(Split(NoSplit, 0))
+        val spaces = style.spaces.inInterpolatedStringCurlyBraces
+        val needed =
+          spaces.isAlways ||
+            spaces.isIfComplex && (
+              rightOwner.children.nonEmpty ||
+                !rightOwner.parent.contains(leftOwner)
+            )
+        Seq(Split(Space(needed), 0))
       case FormatToken(_, close @ T.RightBrace(), _)
           if rightOwner.is[SomeInterpolate] =>
-        Seq(Split(NoSplit, 0))
+        val spaces = style.spaces.inInterpolatedStringCurlyBraces
+        val needed =
+          spaces.isAlways ||
```

Comment:
can you please define this logic as a method in the `SpacesInInterpolatedStringCurlyBraces` and reuse?